tests/Unit/TestCase.php

```diff
@@ -57,54 +85,27 @@ protected function setProtected($obj, string $property, $value): void
     /**
      * Get a protected property of an object.
      *
-     * @param mixed  $obj      object Instance
-     * @param string $property property name
-     *
-     * @return mixed property value
+     * @throws ReflectionException
      */
-    protected function getProtected($obj, string $property)
+    protected function getProtected(object|string $obj, string $property): mixed
     {
         $class = new ReflectionClass($obj);
         $property = $class->getProperty($property);
         $property->setAccessible(true);
+
         return $property->getValue($obj);
     }
 
     /**
      * Replace instance on Ioc
      */
-    protected function instance(string $abstract, $instance)
+    protected function instance(string $abstract, mixed $instance): mixed
```

Comment:
Aqui a instância não seria sempre um objeto?

```suggestion
    protected function instance(string $abstract, object $instance): object
```

Reply:
done!